src/boot/z_std_dma.c

```diff
@@ -367,14 +486,23 @@ s32 DmaMgr_SendRequestImpl(DmaRequest* req, void* ram, uintptr_t vrom, u32 size,
     return 0;
 }
 
-s32 DmaMgr_SendRequest0(void* ram, uintptr_t vrom, u32 size) {
+/**
+ * Submit a synchronous DMA request. This will block the current thread until the requested transfer is complete. Data
+ * is immediately available once the current thread resumes execution.
+ *
+ * @param ram Location in DRAM for data to be written.
+ * @param vrom Virtual ROM location for data to be read.
+ * @param size Transfer size.
+ * @return 0
+ */
+s32 DmaMgr_SyncDmaRequest(void* ram, uintptr_t vrom, size_t size) {
     DmaRequest req;
     OSMesgQueue queue;
     OSMesg msg;
     s32 ret;
 
     osCreateMesgQueue(&queue, &msg, 1);
-    ret = DmaMgr_SendRequestImpl(&req, ram, vrom, size, 0, &queue, NULL);
+    ret = DmaMgr_SendRequest(&req, ram, vrom, size, 0, &queue, NULL);
     if (ret == -1) {
```

Comment:
Not sure if it is worth mentioning the `DmaMgr_SendRequest` cannot return `-1`. The return 0 was confusing at first due to this.